cpp/include/cudf/strings/convert/convert_fixed_point.hpp

```diff
@@ -0,0 +1,124 @@
+ * provided strings column.
+ *
+ * Any null entries will result in corresponding null entries in the output column.
+ *
+ * Only characters [0-9] plus a prefix '-' and '+' and a single decimal point
+ * are recognized. When any other character is encountered, the parsing ends
+ * for that string and the current digits are converted into a fixed-point value.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ['123', '-876', '543.2', '-0.12']
+ * dt = {DECIMAL32, scale=-2}
+ * fp = to_fixed_point(s, dt)
+ * fp is [123400, -87600, 54320, -12]
+ * @endcode
+ *
+ * Overflow of the resulting value type is not checked.
+ * The scale in the `output_type` is used for setting the integer component.
+ *
+ * @throw cudf::logic_error if `output_type` is not a fixed-point decimal type.
+ *
+ * @param strings Strings instance for this operation.
+ * @param output_type Type of fixed-point column to return including the scale value.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New column of `output_type`.
+ */
+std::unique_ptr<column> to_fixed_point(
+  strings_column_view const& input,
+  data_type output_type,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Returns a new strings column converting the fixed-point values
+ * into a strings column.
+ *
+ * Any null entries will result in corresponding null entries in the output column.
+ *
+ * For each value, a string is created in base-10 decimal.
+ * Negative numbers will include a '-' prefix.
+ * The column's scale value is used to place the decimal point.
+ * A negative scale value may add padded zeros after the decimal point.
+ *
+ * @code{.pseudo}
+ * Example:
+ * fp is [110, 222, 3330, -440, -1] with scale = -2
+ * s = from_fixed_point(fp)
+ * s is now ['1.10', '2.22', '33.30', '-4.40', '-0.01']
+ * @endcode
+ *
+ * @throw cudf::logic_error if the `input` column is not a fixed-point decimal type.
+ *
+ * @param input Fixed-point column to convert.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New strings column.
+ */
+std::unique_ptr<column> from_fixed_point(
+  column_view const& input,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Returns a boolean column identifying strings in which all
+ * characters are valid for conversion to fixed-point.
+ *
+ * The output row entry will be set to `true` if the corresponding string element
+ * has at least one character in [+-0123456789.]. The optional sign character
+ * must only be in the first position. The decimal point may only appear once.
+ * Also, the integer component must fit within the size limits of the
+ * underlying fixed-point storage type. The value of the integer component
+ * is based on the scale of the `decimal_type` provided.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ['123', '-456', '', '1.2.3', '+17E30', '12.34' '.789', '-0.005]
```

Comment:
According to the docs for `to_fixed_point`, `1.2.3` is valid and the same as `1.2`, so is it inconsistent that it would return `false` here? Or are the docs for `to_fixed_point` incorrect/incomplete?

Reply:
The docs should say that `to_fixed_point` does not check for valid data format. This is consistent with all other convert functions which do not check format validity -- garbage in / garbage out. String format validity is only checked by the `is_*` functions.

Reply:
I decided to make invalid data format to be undefined behavior `to_fixed_point`. Hopefully this removes the confusion.

Reply:
I think that's the right decision.